src/_TextInputInnerAction.tsx

```diff
@@ -0,0 +1,69 @@
+import {IconProps} from '@primer/octicons-react'
+import React, {forwardRef, MouseEventHandler} from 'react'
+import {Box, Tooltip} from '.'
+import {Button, ButtonProps, IconButton} from './Button'
+import {SxProp} from './sx'
+
+type TextInputActionProps = Omit<React.HTMLProps<HTMLButtonElement>, 'onClick' | 'children' | 'size'> & {
+  children?: React.ReactNode
+  onClick: MouseEventHandler
+  icon?: React.FunctionComponent<IconProps>
+  iconLabel?: string
+  tooltipMessage?: string
+  variant?: ButtonProps['variant']
+} & SxProp
+
+const buttonStyleOverrides = {
+  color: 'fg.default'
+}
+
+// TODO: modify this to _just_ render out a tooltip. No need for `wrapper`
+const ConditionalWrapper: React.FC<{
+  condition: boolean
+  wrapper: (children: React.ReactNode) => React.ReactNode
+}> = ({condition, wrapper, children}) => <>{condition ? wrapper(children) : children}</>
+
+const TextInputAction = forwardRef<HTMLButtonElement, TextInputActionProps>(
+  ({children, icon, iconLabel, sx: sxProp, tooltipMessage, variant, ...rest}, forwardedRef) => {
+    const sx = {
+      ...buttonStyleOverrides,
+      ...sxProp
+    }
```

Comment:
Should we do a deep merge of the sx prop? https://github.com/primer/react/blob/main/src/PageLayout/PageLayout.tsx#L59